base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -62,10 +64,11 @@
  *
  * @author Fraser Tweedale
  */
-public class LDAPDatabase extends ACMEDatabase {
+public class LDAPDatabase extends ACMEDatabase implements Runnable {
```

Comment:
This is the part I don't like. Could we enclose Runnable as an inner class? This would still let you modify the outside instance, and in the future when we get more tasks (we seem to be collect them... :-) ) we can just add another instance versus modifying the existing `run()` instance. I think it is also confusing to see a database with a `Runnable` interface in general.

My 2c.

Reply:
Yeah, I can move it into a separate class.

Reply:
OK, I moved it into `LDAPConfigMonitor` which is in the same package so it can access `LDAPDatabase`'s attributes directly. Please have a look. Thanks!